### ISSUE TITLE
Better error message for not using tables.

### ISSUE THREAD

**Issue by paddyroddy** [pappasam/toml-sort#44]
Taking an example from the [PEP documentation](https://peps.python.org/pep-0621/).

```
[project]
urls.homepage = "https://example.com"
urls.documentation = "https://readthedocs.org"}
```

Running `toml-sort` results in
```
Traceback (most recent call last):
  File "/Users/paddy/mambaforge/bin/toml-sort", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/toml_sort/cli.py", line 297, in cli
    args = get_parser().parse_args(args=arguments)  # strip command itself
           ^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/toml_sort/cli.py", line 289, in get_parser
    parser.set_defaults(**load_config_file())
                          ^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/toml_sort/cli.py", line 89, in load_config_file
    document = tomlkit.parse(content)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/tomlkit/api.py", line 83, in parse
    return Parser(string).parse()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/tomlkit/parser.py", line 158, in parse
    key, value = self._parse_table()
                 ^^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/tomlkit/parser.py", line 999, in _parse_table
    item = self._parse_item()
           ^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/tomlkit/parser.py", line 242, in _parse_item
    return self._parse_key_value(True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/tomlkit/parser.py", line 338, in _parse_key_value
    cws, comment, trail = self._parse_comment_trail()
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/tomlkit/parser.py", line 284, in _parse_comment_trail
    raise self.parse_error(UnexpectedCharError, c)
tomlkit.exceptions.UnexpectedCharError: Unexpected character: '}' at line 3 col 46
```

This can be modified to
```
[project]
urls = {homepage = "https://example.com", documentation = "https://readthedocs.org"}
```
and now `toml-sort` will work as expected. However, from the error message, it is not at all obvious that this is what is required.

**Comment by jonathangreen** [comment]
@paddyroddy I submitted a PR (#45) that I believe will resolve this issue. 

In the toml you included in this issue, there is a indeed an unexpected trailing `}` character, which makes the toml invalid.

However even without that trailing character this toml:
```toml
[project]
urls.homepage = "https://example.com"
urls.documentation = "https://readthedocs.org"
```

Was causing a `TypeError` exception, which was a bug. This bug should be resolved once #45 is merged.

**Comment by paddyroddy** [comment]
Apologies, you are right. What I meant to say (and the error I had seen before) was more like this:
```
[project]
urls.homepage = "https://example.com"
urls.documentation = "https://readthedocs.org"
```


```
Traceback (most recent call last):
  File "/Users/paddy/mambaforge/bin/toml-sort", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/toml_sort/cli.py", line 356, in cli
    ).sorted()
      ^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/toml_sort/tomlsort.py", line 636, in sorted
    sorted_toml = self.toml_doc_sorted(toml_doc)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/toml_sort/tomlsort.py", line 622, in toml_doc_sorted
    item.key, self.toml_elements_sorted(item, sorted_document)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/toml_sort/tomlsort.py", line 441, in toml_elements_sorted
    item.key, self.toml_elements_sorted(item, previous_item)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/toml_sort/tomlsort.py", line 438, in toml_elements_sorted
    previous_item = self.table_previous_item(new_table, parent)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/paddy/mambaforge/lib/python3.11/site-packages/toml_sort/tomlsort.py", line 478, in table_previous_item
    raise TypeError(
TypeError: Unexpected type <class 'tomlkit.items.String'>, cannot insert comment.
```
